sdk/rust/src/processor/handler.rs

```diff
@@ -0,0 +1,243 @@
+        match *self {
+            ContextError::AuthorizationError(ref msg) => msg,
+            ContextError::UnknownError(ref msg) => msg,
+            ContextError::SerializationError(ref err) => err.description(),
+            ContextError::SendError(ref err) => err.description(),
+            ContextError::ReceiveError(ref err) => err.description(),
+        }
+    }
+
+    fn cause(&self) -> Option<&std::error::Error> {
+        match *self {
+            ContextError::AuthorizationError(_) => None,
+            ContextError::UnknownError(_) => None,
+            ContextError::SerializationError(ref err) => Some(err.borrow()),
+            ContextError::SendError(ref err) => Some(err.borrow()),
+            ContextError::ReceiveError(ref err) => Some(err.borrow()),
+        }
+    }
+}
+
+impl std::fmt::Display for ContextError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            ContextError::AuthorizationError(ref s) =>
+                write!(f, "AuthorizationError: {}", s),
+            ContextError::UnknownError(ref s) =>
+                write!(f, "UnknownError: {}", s),
+            ContextError::SerializationError(ref err) =>
+                write!(f, "SerializationError: {}", err.description()),
+            ContextError::SendError(ref err) =>
+                write!(f, "SendError: {}", err.description()),
+            ContextError::ReceiveError(ref err) =>
+                write!(f, "ReceiveError: {}", err.description()),
+        }
+    }
+}
+
+impl From<ContextError> for ApplyError {
+    fn from(context_error: ContextError) -> Self {
+        ApplyError::InvalidTransaction(format!("{}", context_error))
+    }
+}
+
+impl From<protobuf::ProtobufError> for ContextError {
+    fn from(e: protobuf::ProtobufError) -> Self {
+        ContextError::SerializationError(Box::new(e))
+    }
+}
+
+impl From<SendError> for ContextError {
+    fn from(e: SendError) -> Self {
+        ContextError::SendError(Box::new(e))
+    }
+}
+
+impl From<ReceiveError> for ContextError {
+    fn from(e: ReceiveError) -> Self {
+        ContextError::ReceiveError(Box::new(e))
+    }
+}
+
+pub struct TransactionContext {
+    context_id: String,
+    sender: ZmqMessageSender
+}
+
+impl TransactionContext {
+    pub fn new(context_id: &str, sender: ZmqMessageSender) -> TransactionContext {
+        TransactionContext{
+            context_id: String::from(context_id),
+            sender: sender
+        }
+    }
+
+    pub fn get_state(&mut self, address: &str) -> Result<Option<Vec<u8>>, ContextError> {
+        let mut request = TpStateGetRequest::new();
+        request.set_context_id(self.context_id.clone());
+        request.set_addresses(RepeatedField::from_vec(vec!(String::from(address))));
+        let serialized = request.write_to_bytes()?;
+        let x : &[u8] = &serialized;
+
+        let correlation_id: String = rand::thread_rng().gen_ascii_chars().take(16).collect();
```

Comment:
In other implementations, correlation id generation is behind another function, which is what I would expect here. If not, at a minimum 16 should be moved to a constant.

Reply:
This is now a function generate_correlation_id().